### PR TITLE
New id generation algorithm

### PR DESCRIPTION
**Pull request by kof**

### CODE REVIEW

File: tests/integration/sheet.js [hunk 1]
```diff
@@ -287,12 +275,12 @@ describe('Integration: sheet', () => {
...
       expect(sheet.toString()).to.be(
         '@media print {\n' +
-        '  .a-3787690649 {\n' +
+        '  .my-special-id {\n' +
         '    float: left;\n' +
         '  }\n' +
         '}\n' +
         '@media screen {\n' +
-        '  .a-3787690649 {\n' +
+        '  .my-special-id {\n' +
```
Comment (iamstarkov) [inline tests/integration/sheet.js]: i believe this usecase should end up with different ids in real world

Comment (kof) [inline tests/integration/sheet.js]: Why, this case should have ensured actually the opposite. But it was late and I made a mistake with a fixed id. Now we don't verify that the id is really same.

Comment (iamstarkov) [inline tests/integration/sheet.js]: why do you want this ids to be the same?

Comment (iamstarkov) [inline tests/integration/sheet.js]: ok, you are right

Comment (iamstarkov) [inline tests/integration/sheet.js]: thanks for explanation

File: tests/integration/sheet.js [hunk 2]
```diff
@@ -255,14 +259,14 @@ describe('Integration: sheet', () => {
...
-          `  }\n` +
-          `}\n` +
+          '    float: left;\n' +
+          '  }\n' +
+          '}\n' +
           `.${id} {\n` +
-          `  color: red;\n` +
-          `}`
+          '  color: red;\n' +
+          '}'
```
Comment (iamstarkov) [inline tests/integration/sheet.js]: you may be interested in [`stripindent`](https://www.npmjs.com/package/common-tags#stripindent) literal tag from ['common-tags`](https://www.npmjs.com/package/common-tags) project

Comment (kof) [inline tests/integration/sheet.js]: yep, already started using it

Comment (kof) [inline tests/integration/sheet.js]: need a migration script to rewrite them all

Comment (iamstarkov) [inline tests/integration/sheet.js]: i dont know how did i manage to comment on outdated code. github just showed me latest changes

File: src/utils/generateClassName.js [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
-export default function generateClassName(str: string, rule: Rule, sheet?: StyleSheet): string {
-  if (sheet && sheet.options.meta) str += sheet.options.meta
-
-  const hash = createHash(str)
-
-  // There is no name if `jss.createRule(style)` was used.
-  return rule.name ? `${rule.name}-${hash}` : hash
-}
+export default (str: string, rule: Rule): string => (
+  // There is no rule name if `jss.createRule(style)` was used.
```
Comment (kof) [inline src/utils/generateClassName.js]: actually need to change this interface as we will have no str param any more

**Comment by kof** [comment]
@cvle @iamstarkov can you guys try this branch without calling rehydrate? I have tried just now, seems to be perfectly fine.

**Comment by cvle** [comment]
@kof I'm not running SSR anywhere atm except on an old repository of mine, let's see if I can revive it.

**Comment by kof** [comment]
In order to move forward with this, I will remove rehydration and release a minor version to receive some feedbacks.

**Comment by kof** [comment]
btw. @oliviertassinari maybe material-ui has some server-side tests we can run to make sure this doesn't break anything?

**Comment by oliviertassinari** [comment]
@kof I wish we add some server-side tests but as far as I know, we don't. That's definitely something we should add!

**Pull request merged**